src/spec-node/collectionCommonUtils/outdatedCommandImpl.ts

```diff
@@ -0,0 +1,441 @@
+
+		const outputParams = { output, env: process.env };
+		const dockerParams = await createDockerParams({
+			containerDataFolder: undefined,
+			containerSystemDataFolder: undefined,
+			workspaceFolder,
+			mountWorkspaceGitRoot: false,
+			configFile,
+			logLevel: mapLogLevel(logLevel),
+			logFormat,
+			log: text => process.stderr.write(text),
+			terminalDimensions: /* terminalColumns && terminalRows ? { columns: terminalColumns, rows: terminalRows } : */ undefined, // TODO
+			defaultUserEnvProbe: 'loginInteractiveShell',
+			removeExistingContainer: false,
+			buildNoCache: false,
+			expectExistingContainer: false,
+			postCreateEnabled: false,
+			skipNonBlocking: false,
+			prebuild: false,
+			persistedFolder: undefined,
+			additionalMounts: [],
+			updateRemoteUserUIDDefault: 'never',
+			additionalCacheFroms: [],
+			useBuildKit: 'never',
+			buildxPlatform: undefined,
+			buildxPush: false,
+			buildxOutput: undefined,
+			buildxCacheTo: undefined,
+			skipFeatureAutoMapping: false,
+			skipPostAttach: false,
+			skipPersistingCustomizationsFromFeatures: false,
+			dotfiles: {
+				repository: undefined,
+				installCommand: undefined,
+				targetPath: undefined
+			},
+			dockerPath: undefined,
+			dockerComposePath: undefined,
+			overrideConfigFile: undefined,
+			remoteEnv: {}
+		}, disposables);
+
+		const outdatedImages = await loadImageVersionInfo(outputParams, configs.config.config, cliHost, dockerParams);
+
+		await new Promise<void>((resolve, reject) => {
+			let text = '';
+			if (outputFormat === 'text') {
+				const featureRows = Object.keys(outdatedFeatures.features).map(key => {
+					const value = outdatedFeatures.features[key];
+					return [getFeatureIdWithoutVersion(key), value.current, value.wanted, value.latest]
+						.map(v => v === undefined ? '-' : v);
+				});
+
+				if (featureRows.length !== 0) {
+					const featureHeader = ['Feature', 'Current', 'Wanted', 'Latest'];
+					text = textTable([
+						featureHeader,
+						...featureRows,
+					]);
+				}
+
+				const imageRows = Object.keys(outdatedImages.images).map(key => {
+					const value = outdatedImages.images[key];
+					return [value.name, value.current, value.wanted]
+						.map(v => v === undefined ? '-' : v);
+				});
+
+				if (imageRows.length !== 0) {
+					const imageHeader = ['Image', 'Current', 'Latest'];
+					text += '\n\n';
+					text += textTable([
+						imageHeader,
+						...imageRows,
+					]);
+				}
+			} else {
+				text = JSON.stringify({ ...outdatedFeatures, ...outdatedImages }, undefined, process.stdout.isTTY ? '  ' : undefined);
+			}
+			process.stdout.write(text + '\n', err => err ? reject(err) : resolve());
+		});
+	} catch (err) {
+		if (output) {
+			output.write(err && (err.stack || err.message) || String(err));
+		} else {
+			console.error(err);
+		}
+		await dispose();
+		process.exit(1);
+	}
+	await dispose();
+	process.exit(0);
+}
+
+/*
```

Comment:
You could add an interface for this type, that would then help avoiding the use of `any` for the maps where the info objects are collected at the calling function.

Reply:
Updated!